src/test/java/core/basesyntax/service/impl/StorageServiceImplTest.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StorageServiceImplTest {
+
+    private StorageServiceImpl storageService;
+    private Fruit apple;
+
+    @BeforeEach
+    void setUp() {
+        storageService = new StorageServiceImpl();
+        apple = new Fruit("apple");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.clear();
+    }
+
+    @Test
+    void addFruit_validStorageUpdating_ok() {
+        storageService.addFruit(apple, 50);
+        assertEquals(50, Storage.getFruitQuiantity(apple));
+    }
+
+    @Test
+    void removeFruit_validStorageUpdating_ok() {
+        storageService.addFruit(apple, 50);
+        storageService.removeFruit(apple, 20);
```

Comment:
Don't use the same method or service to prepare storage. What are you testing here addFruit or removeFruit?

Reply:
This test was writen to ensure that removeFruit() correctly updates fruit's quantity